### DISCUSSION TITLE
How to generate all entire CSS rules?

### DISCUSSION THREAD

**Discussion by PrinOrange** [tailwindlabs/tailwindcss#9006] [Help] [answered]
I'm using Tailwind CSS in Electron.js, not in the browser. So I don't need to minify the generated CSS file or generate CSS on demand. So I want to generate all entire CSS classes into one file. But I checked the official documentation and it doesn't seem to tell me how to generate a full CSS file. So I went to the community for answers.
The solution they gave me was add the `safelist` rule in the configuration, `safelist`
```
  safelist: [
    {
      pattern: /.*/,
    },
  ],
```
But it doesn't generate CSS rules for pseudo-classes. For example, when I use the class `hover:text-white` or `after:bg-white` in my DOM, it won't work. That is to say, it does not generate corresponding CSS rules for pseudo-classes such as hover, after, etc. in the file.
So, I sincerely ask, how to generate all the CSS rules? Thank you so much.

**Comment by PrinOrange** [comment]
🙏 I'm troubled with such problems. Thanks for every one

**Comment by wongjn** [comment]
As per the [documentation of safelisting with regular expressions](https://tailwindcss.com/docs/content-configuration#using-regular-expressions), there is a `variants` key that you can use along side `pattern`. So to add the `hover` variants, it would look like this:

```
safelist: [
  {
    pattern: /.*/,
    variants: ['hover'],
  },
],
```

**Reply by PrinOrange** [reply]
Thanks for your answer very much, this also solved my problem to some extent

**Comment by adamwathan** [comment, marked as answer]
There’s simply no such thing as a “full” version of Tailwind 3 — the “full” version would be probably 100gb or more because of how many variants we support and the fact that they can be stacked together, and wouldn’t include many of the things that Tailwind can do that can’t be determined ahead of time, like arbitrary values and arbitrary variants. You definitely want to just generate only what you need, even with Electron, otherwise the dev tools in Electron will slow down to a halt and run-time rendering performance will suffer.

**Reply by PrinOrange** [reply]
Thank you very much!
